src/gatsby-node.js

```diff
@@ -248,3 +250,37 @@ exports.createResolvers = ({
   });
   createResolvers(resolvers);
 };
+
+exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    api_key: Joi.string().required().description(`API Key is a unique key assigned to each stack.`),
+    delivery_token: Joi.string().required().description(`Delivery Token is a read-only credential.`),
+    environment: Joi.string().required().description(`Environment where you published your data.`),
+    cdn: Joi.string().default(`https://cdn.contentstack.io/v3`).description(`CDN set this to point to other cdn end point. For eg: https://eu-cdn.contentstack.com/v3 `),
+    type_prefix:  Joi.string().default(`Contentstack`).description(`Specify a different prefix for types. This is useful in cases where you have multiple instances of the plugin to be connected to different stacks.`),
+    expediteBuild: Joi.boolean().default(false).description(`expediteBuild set this to either true or false.`),
+    enableSchemaGeneration: Joi.boolean().default(false).description(`Specify true if you want to generate custom schema.`),
+  }).external(validateContentstackAccess)
+}
+
+
+const validateContentstackAccess = async pluginOptions => {
+  if (process.env.NODE_ENV === `test`) return undefined
+  
+  let host = pluginOptions.cdn ? pluginOptions.cdn : 'https://cdn.contentstack.io/v3';
```

Comment:
This line can be changed to
`const host = pluginOptions.cdn || 'https://cdn.contentstack.io/v3';`
changing from let to const as value of host never changes also using || operator to get selective value instead of ternary operator.